File: go/clients/p2p/examples/WebSocket/WsChat.go
package main

import (
	"context"
	"fmt"
	"os"
	"os/signal"
	"syscall"
	"time"

	"github.com/mexcdevelop/mexc-api-demo/go/clients/p2p"
)

func main() {
	listenKey := os.Getenv("MEXC_P2P_LISTEN_KEY")
	conversationID := os.Getenv("MEXC_P2P_CONVERSATION_ID")
	if listenKey == "" || conversationID == "" {
		fmt.Println("Set MEXC_P2P_LISTEN_KEY and MEXC_P2P_CONVERSATION_ID")
		os.Exit(1)
	}

	ws := p2p.NewChatWSClient(p2p.ChatWSConfig{
		ListenKey: listenKey, ConversationID: conversationID,
	})
	ws.OnMessage = func(msg p2p.ChatMessage) {
		fmt.Println(msg.FromNickName, msg.Content)
	}
	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
	defer cancel()
	if err := ws.Connect(ctx); err != nil {
		fmt.Println(err)
		return
	}
	sig := make(chan os.Signal, 1)
	signal.Notify(sig, syscall.SIGINT, syscall.SIGTERM)
	<-sig
	ws.Disconnect()
}

File: node.js/clients/p2p/examples/WebSocket/WsChat.js
/**
 * Private WebSocket — P2P chat
 * API docs: P2P API → Chat / User Data Stream
 * Run: node examples/WebSocket/WsChat.js
 */
const { P2PChatWsClient } = require('../_lib')

const listenKey = process.env.MEXC_P2P_LISTEN_KEY || ''
const conversationId = process.env.MEXC_P2P_CONVERSATION_ID || ''

if (!listenKey || !conversationId) {
  console.error('Set MEXC_P2P_LISTEN_KEY and MEXC_P2P_CONVERSATION_ID in .env')
  process.exit(1)
}

const client = new P2PChatWsClient({ listenKey, conversationId })
client.on('open', () => console.log('connected'))
client.on('message', (msg) => console.log('[chat]', msg.fromNickName, msg.content))
client.on('error', (err) => console.error(err?.message ?? err))
client.connect()
process.on('SIGINT', () => { client.disconnect(); process.exit(0) })
